### PR TITLE
Feature/movies data layer

### PR DESCRIPTION
**Pull request by scadavidg**

### CODE REVIEW

**Comment by ccruzt** [comment]
Hi @scadavidg , please resolve conflicts

File: movies/data/src/main/java/com/example/movies/data/di/DataSourceModule.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import com.example.movies.data.datasource.MoviesRemoteDataSourceImpl
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object DataSourceModule {
```
Comment (ManuelUrbanoGlobant) [inline movies/data/src/main/java/com/example/movies/data/di/DataSourceModule.kt]: remove this module and add Provide to MoviesDataModule

File: movies/data/src/main/java/com/example/movies/data/di/DataSourceModule.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    @Singleton
+    @Provides
+    fun providesMoviesRemoteDataSource(moviesService: MoviesService): MoviesRemoteDataSource {
+        return MoviesRemoteDataSourceImpl(moviesService, BuildConfig.API_KEY)
```
Comment (ManuelUrbanoGlobant) [inline movies/data/src/main/java/com/example/movies/data/di/DataSourceModule.kt]: Please, convert to expression body

File: movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import com.example.movies.data.mappers.MovieMapper
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object DtoMapperModule {
```
Comment (ManuelUrbanoGlobant) [inline movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt]: same here

File: movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    @Singleton
+    @Provides
+    fun provideMovieMapper(): MovieMapper {
+        return MovieMapper()
```
Comment (ManuelUrbanoGlobant) [inline movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt]: Please convert expression body

File: movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    @Singleton
+    @Provides
+    fun provideMovieDetailMapper(): MovieDetailMapper {
+        return MovieDetailMapper()
```
Comment (ManuelUrbanoGlobant) [inline movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt]: same here

File: movies/data/src/main/java/com/example/movies/data/di/RepositoryModule.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import com.example.movies.domain.repositories.MoviesRepository
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object RepositoryModule {
```
Comment (ManuelUrbanoGlobant) [inline movies/data/src/main/java/com/example/movies/data/di/RepositoryModule.kt]: same here

File: movies/data/src/main/java/com/example/movies/data/di/RepositoryModule.kt [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    @Singleton
+    @Provides
+    fun providesMoviesRepository(
+        moviesRemoteDataSource: MoviesRemoteDataSource,
+        movieMapper: MovieMapper,
+        movieDetailMapper: MovieDetailMapper
+    ): MoviesRepository {
+        return MoviesRepositoryImpl(
```
Comment (ManuelUrbanoGlobant) [inline movies/data/src/main/java/com/example/movies/data/di/RepositoryModule.kt]: same here

File: movies/data/src/main/java/com/example/movies/data/repositories/MoviesRepositoryImpl.kt [hunk 8]
```diff
@@ -1,16 +1,47 @@
...
+            val body = response.body()
+            if (body != null && response.isSuccessful) {
+                movieList = movieMapper.fromEntityList(body.movies)
+            } else {
+                return Response.Error("something went wrong")
+            }
+        } catch (error: Exception) {
+            return Response.Error(error.message.toString())
+        }
+        return Response.Success(movieList)
```
Comment (ManuelUrbanoGlobant) [inline movies/data/src/main/java/com/example/movies/data/repositories/MoviesRepositoryImpl.kt]: this function should be,

```
return try {
            val response = moviesRemoteDataSource.getListMovies()
            val body = response.body()
            if (body != null && response.isSuccessful) {
                Response.Success(movieMapper.fromEntityList(body.movies))
            } else {
                Response.Error("something went wrong")
            }
        } catch (error: Exception) {
            Response.Error(error.message.toString())
        }
```

File: movies/data/src/main/java/com/example/movies/data/repositories/MoviesRepositoryImpl.kt [hunk 9]
```diff
@@ -1,16 +1,47 @@
...
+            val body = response.body()
+            if (body != null && response.isSuccessful) {
+                movieDetail = movieDetailMapper.mapFromEntity(body)
+            } else {
+                return Response.Error("something went wrong")
+            }
+        } catch (error: Exception) {
+            return Response.Error(error.message.toString())
+        }
+        return Response.Success(movieDetail)
```
Comment (ManuelUrbanoGlobant) [inline movies/data/src/main/java/com/example/movies/data/repositories/MoviesRepositoryImpl.kt]: I think we can modify in the same way here, please review it.

**Review by ManuelUrbanoGlobant — CHANGES_REQUESTED** [review]

File: movies/data/src/main/java/com/example/movies/data/di/DataSourceModule.kt (hunk 1 above)
Comment (scadavidg) [reply movies/data/src/main/java/com/example/movies/data/di/DataSourceModule.kt]: done

File: movies/data/src/main/java/com/example/movies/data/di/DataSourceModule.kt (hunk 2 above)
Comment (scadavidg) [reply movies/data/src/main/java/com/example/movies/data/di/DataSourceModule.kt]: done

File: movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt (hunk 3 above)
Comment (scadavidg) [reply movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt]: done

File: movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt (hunk 4 above)
Comment (scadavidg) [reply movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt]: done

File: movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt (hunk 5 above)
Comment (scadavidg) [reply movies/data/src/main/java/com/example/movies/data/di/DtoMapperModule.kt]: done

File: movies/data/src/main/java/com/example/movies/data/di/RepositoryModule.kt (hunk 6 above)
Comment (scadavidg) [reply movies/data/src/main/java/com/example/movies/data/di/RepositoryModule.kt]: done

File: movies/data/src/main/java/com/example/movies/data/di/RepositoryModule.kt (hunk 7 above)
Comment (scadavidg) [reply movies/data/src/main/java/com/example/movies/data/di/RepositoryModule.kt]: done

File: movies/data/src/main/java/com/example/movies/data/repositories/MoviesRepositoryImpl.kt (hunk 8 above)
Comment (scadavidg) [reply movies/data/src/main/java/com/example/movies/data/repositories/MoviesRepositoryImpl.kt]: done

File: movies/data/src/main/java/com/example/movies/data/repositories/MoviesRepositoryImpl.kt (hunk 9 above)
Comment (scadavidg) [reply movies/data/src/main/java/com/example/movies/data/repositories/MoviesRepositoryImpl.kt]: done

**Review by ManuelUrbanoGlobant — APPROVED** [review]

**Pull request merged**